src/Plugins/ApplicationLogs/Store/LogStorageStore.cs

```diff
@@ -159,9 +160,13 @@ public Guid PutStackItemState(StackItem stackItem)
             {
                 _snapshot.Put(key, BinarySerializer.Serialize(stackItem, ExecutionEngineLimits.Default with { MaxItemSize = (uint)Settings.Default.MaxStackSize }));
             }
-            catch (NotSupportedException)
+            catch (Exception ex)
             {
-                _snapshot.Put(key, BinarySerializer.Serialize(StackItem.Null, ExecutionEngineLimits.Default with { MaxItemSize = (uint)Settings.Default.MaxStackSize }));
+                var detailedException = ex.Message;
+                if (detailedException.Length > 1024)
```

Comment:
Null or exception both works. Using `exception message` to explicitly tell people there is an issue.

Reply:
We think that it's not quite correct to replace the original stackitem by either `Null` or `ByteString` because from the user side it's impossible to distinguish the "good" original stackitem from the result of invalid seialization.

Please, take a look at the way how this case is handled in NeoGo: https://github.com/nspcc-dev/neo-go/blob/cf4d4a2611209d8d085ccec6e1f1b8956006a577/pkg/vm/stackitem/serialization.go#L144. We use a special stackitem of `InvalidT` type which can't be produced by any means other than exception during serialisation. With this special type there's no doubts from the user side whether there was an exception during serialization or not.

Reply:
> We think that it's not quite correct to replace the original stackitem by either Null or ByteString because from the user side it's impossible to distinguish the "good" original stackitem from the result of invalid seialization.


i understand your concern, but issues has being there and this pr is not for that purpose, you can have another pr to fix it. for me, no one but hackers will ever see that information.

Reply:
But if we're changing this code anyway, why not to make it work properly?

Reply:
> But if we're changing this code anyway, why not to make it work properly?

Cause its not the purpose of this pr, though its easy to update, but that is your idea and your property, and most importantly, its another issue.

Reply:
I agree with the fix itself. But then let's not to change this line:
```
_snapshot.Put(key, BinarySerializer.Serialize(StackItem.Null, ExecutionEngineLimits.Default with { MaxItemSize = (uint)Settings.Default.MaxStackSize }));
```

Reply:
Ref. https://github.com/neo-project/neo/issues/3303.